### PR TITLE
add saving safety_checker

### PR DESCRIPTION
**Pull request by eaidova**
# What does this PR do?

safety_checker is part of pipeline for sd and sdxl models family. In case, if we provide already converted model and model_index.json contains safety_checker object definition, pipeline loading failed in attempt to load model for it:
OSError: Error no file named pytorch_model.bin, model.safetensors, tf_model.h5, model.ckpt.index or flax_model.msgpack found in directory

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/intel/openvino/modeling_diffusion.py [hunk 1]
```diff
@@ -422,12 +424,15 @@ def _from_pretrained(
                     module = getattr(pipelines, module_name)
                 else:
                     module = importlib.import_module(module_name)
+                logger.warn(module)
```
Comment (eaidova) [inline optimum/intel/openvino/modeling_diffusion.py:427]:
```suggestion
```

File: optimum/intel/openvino/modeling_diffusion.py [hunk 2]
```diff
@@ -422,12 +424,15 @@ def _from_pretrained(
...
                 class_obj = getattr(module, module_class)
                 load_method = getattr(class_obj, "from_pretrained")
                 # Check if the module is in a subdirectory
                 if (model_save_path / name).is_dir():
+                    logger.warn(name)
```
Comment (eaidova) [inline optimum/intel/openvino/modeling_diffusion.py:432]:
```suggestion

```

File: optimum/intel/openvino/modeling_diffusion.py [hunk 3]
```diff
@@ -422,12 +424,15 @@ def _from_pretrained(
...
                     submodels[name] = load_method(model_save_path / name)
                 else:
+                    logger.warn(name)
```
Comment (eaidova) [inline optimum/intel/openvino/modeling_diffusion.py:435]:
```suggestion
```

File: optimum/intel/openvino/modeling_diffusion.py [hunk 4]
```diff
@@ -449,8 +454,10 @@ def _from_pretrained(
         if (quantization_config is None or quantization_config.dataset is None) and not compile_only:
             for name, path in models.items():
                 if name in kwargs:
+                    logger.warn(name)
```
Comment (eaidova) [inline optimum/intel/openvino/modeling_diffusion.py:457]:
```suggestion
```

File: optimum/intel/openvino/modeling_diffusion.py [hunk 5]
```diff
@@ -449,8 +454,10 @@ def _from_pretrained(
...
                     models[name] = kwargs.pop(name)
                 else:
+                    logger.warn(name)
```
Comment (eaidova) [inline optimum/intel/openvino/modeling_diffusion.py:460]:
```suggestion
```

File: optimum/exporters/openvino/convert.py [hunk 6]
```diff
@@ -715,6 +715,9 @@ def export_from_model(
         tokenizer_3 = getattr(model, "tokenizer_3", None)
         if tokenizer_3 is not None:
             tokenizer_3.save_pretrained(output.joinpath("tokenizer_3"))
+        safety_checker = getattr(model, "safety_checker", None)
+        if safety_checker is not None:
+            safety_checker.save_pretrained(output.joinpath("safety_checker"))
```
Comment (echarlaix) [inline optimum/exporters/openvino/convert.py:720]: could you add a test to make sure the `safety_checker` is correctly saved and can later be loaded ?

**Review by echarlaix — COMMENTED** [review]
LGTM!

File: optimum/exporters/openvino/convert.py (hunk 6 above)
Comment (eaidova) [reply optimum/exporters/openvino/convert.py:720]: added

**Review by nikita-savelyevv — APPROVED** [review]

**Review by AlexKoff88 — APPROVED** [review]

**Pull request merged**